_posts/2021-01-22-x-ways-to-write-meaningful-commit-messages.md

```diff
@@ -0,0 +1,140 @@
+
+## What is a good commit message?
+
+Writing suitable commit messages can be defined as many styles. The trick is to pick the best that suits you and then stick to it. It means being **consistent** in using the format in all commits.
+
+The perfect commit message should have certain qualities:
+  - **Understandable** even by seeing only the header of the message (We'll talk about the header soon)
+  - Not too much detailed, but **just enough**
+  - **Unambiguous**
+
+Writing suitable commit messages can be defined as many models. The trick is to pick the best that suits you and then stick to it. It means being consistent in using the format in all commits.
+
+There are many styles we can follow to reach these standards. Angular suggested style is one of them that we're going to explain a simplified version of it later in this article.
+
+All styles and our commits should have some specific qualities to form a structured commit message which we’re going to explain now.
+
+### Atomic Commit:
+Although using a proper style is a good practice, but it's not enough. The discipline is crucial. The commits should be **reasonable and atomic**. If the commit consists of multiple changes that make the message too long or inefficient, it's good practice to **separate it into several commits**. In other words, you don't want to commit a code that does nothing or does too much, or maybe it's broken.
+
+If you commit two changes together, for example, a bug fix, and a minor refactoring, It might not cause a very long commit message, but it can cause some problems. Let's say the bug fix made some other bugs. In that case, you should roll back to your recent commit. It results in the loss of that refactoring as well. It's not efficient, and It's not atomic.
+
+### Short and Unambiguous:
+The message should describe what changes your code makes to the base code. It should answer the question: **"What happens if the changes are applied?"**. If you can not be short, it might be because the commit is not atomic, and it's too much change in one commit.
+
+#### Active Voice:
+ Use the **imperative, present tense**. It is easier to read and skim.
+> **Right**: Add feature for a user to upload pictures
+>
+> **Wrong**: Added feature ... (passive voice, past tense)
+
+#### Detailed Enough:
+Super-detailed commit messages are frustrated as well. You can find that much detail in the system. For example, If your version control is Git, you can see all the changed files in Git. So instead of answering "what are the changes? ", it's better to answer **"What are the changes for?"**.
+
+### Formatting:
+Bellow, we describe a simplified format of what Angular suggests.
+This format consists of a header, a body, and a footer with a blank line between each section.
+
+Header: Mandatory - Up to 50 characters (as Git suggests)
```

Comment:
"as Git suggests" ... what else does Git suggest? Why don't we add a section before this one that describes the Git commit message standard? That would make this article more complete.